Sources/MapboxCoreNavigation/NavigationService.swift

```diff
@@ -182,26 +208,45 @@ public class MapboxNavigationService: NSObject, NavigationService {
     private var _simulationSpeedMultiplier: Double = 1.0
     
     private func simulate(intent: SimulationIntent = .manual) {
-        guard !isSimulating else { return }
         let progress = router.routeProgress
         delegate?.navigationService(self, willBeginSimulating: progress, becauseOf: intent)
+        announceSimulatingDidChange(simulatingUpdate: .willBeginSimulating)
+        guard !isSimulating else {
```

Comment:
Because when CarPlay is used and under simulated mode, the `navigationService.simulate(intent:)` would be called twice. When we choose to start navigation from mobile, then the previous `guard !isSimulating else { return }` before the notification would block sending notification to the CarPlay.